packages/netlify-cms-widget-markdown/src/MarkdownControl/RawEditor.js

```diff
@@ -40,39 +42,55 @@ export default class RawEditor extends React.Component {
     return !this.state.value.equals(nextState.value);
   }
 
-  handleChange = change => {
-    if (!this.state.value.document.equals(change.value.document)) {
-      this.handleDocumentChange(change);
+  handleCopy = (event, editor) => {
+    const { getAsset, resolveWidget } = this.props;
+    const markdown = Plain.serialize(editor.value);
+    const html = markdownToHtml(markdown, { getAsset, resolveWidget });
+    setEventTransfer(event, 'text', markdown);
+    setEventTransfer(event, 'html', html);
+    event.preventDefault();
+  };
+
+  handleCut = (event, editor, next) => {
+    this.handleCopy(event, editor, next);
+    editor.delete();
+  };
+
+  handlePaste = (event, editor, next) => {
+    const data = event.clipboardData;
+    if (isHotkey('shift', event)) {
+      return next();
     }
-    this.setState({ value: change.value });
+
+    const value = Plain.deserialize(data.getData('text/plain'));
+    return editor.insertFragment(value.document);
+  };
+
+  handleChange = editor => {
+    console.log('change');
+    if (!this.state.value.document.equals(editor.value.document)) {
+      this.handleDocumentChange(editor);
+    }
+    this.setState({ value: editor.value });
   };
 
   /**
    * When the document value changes, serialize from Slate's AST back to plain
    * text (which is Markdown) and pass that up as the new value.
    */
-  handleDocumentChange = debounce(change => {
-    const value = Plain.serialize(change.value);
+  handleDocumentChange = debounce(editor => {
+    console.log('document change');
+    const value = Plain.serialize(editor.value);
+    console.log('serialized');
     this.props.onChange(value);
   }, 150);
 
-  /**
-   * If a paste contains plain text, deserialize it to Slate's AST and insert
-   * to the document. Selection logic (where to insert, whether to replace) is
-   * handled by Slate.
-   */
-  handlePaste = (e, data, change) => {
-    if (data.text) {
-      const fragment = Plain.deserialize(data.text).document;
-      return change.insertFragment(fragment);
-    }
-  };
-
   handleToggleMode = () => {
     this.props.onMode('visual');
   };
 
   render() {
+    console.log('render');
```

Comment:
Are these `console.log` intentional?

Reply:
Yeah it’s not merge ready yet, still a little more to do.